.drone.yml

```diff
@@ -11,188 +12,278 @@ workspace:
+
+workspace:
+  base: /go
+  path: src/github.com/gitpods/gitpods
+
+steps:
+- name: binary
   image: golang:1.10
+  pull: always
+  environment:
+    CGO_ENABLED: 0
   commands:
-  - "CGO_ENABLED=0 go build -v -ldflags '-w -extldflags '-static'' -o ./cmd/storage/storage ./cmd/storage"
-  settings:
-    group: build
-
-# Needs to be enabled with dart 2 again
-
-#- name: ui
-#  pull: default
-#  image: golang:1.10
-#  commands:
-#  - "curl -s -L https://github.com/gobuffalo/packr/releases/download/v1.10.4/packr_1.10.4_linux_amd64.tar.gz | tar -xz"
-#  - ./packr
-#  - "CGO_ENABLED=0 go build -v -ldflags '-w -extldflags '-static'' -o ./cmd/ui/ui ./cmd/ui"
-#  settings:
-#    group: build
-
-- name: docker-api
-  pull: default
-  image: plugins/docker
+  - go build -v -ldflags '-w -extldflags '-static'' -o ./cmd/api/api ./cmd/api
+
+- name: docker
+  image: plugins/docker:17.12
+  pull: always
   settings:
+    username:
+      from_secret: docker_username
+    password:
+      from_secret: docker_password
     dockerfile: ./cmd/api/Dockerfile
-    group: docker
     repo: gitpods/api
     tag:
     - latest
-  environment:
-    docker_password:
-      from_secret: docker_password
-    docker_username:
-      from_secret: docker_username
   when:
     branch:
     - master
     event:
     - push
 
-- name: docker-storage
-  pull: default
-  image: plugins/docker
+depends_on:
+- testing
+
+---
+kind: pipeline
+name: storage
+
+platform:
+  os: linux
+  arch: amd64
+
+workspace:
+  base: /go
+  path: src/github.com/gitpods/gitpods
+
+steps:
+- name: binary
+  image: golang:1.10
+  pull: always
+  environment:
+    CGO_ENABLED: 0
+  commands:
+  - go build -v -ldflags '-w -extldflags '-static'' -o ./cmd/storage/storage ./cmd/storage
+
+- name: docker
+  image: plugins/docker:17.12
+  pull: always
   settings:
-    context: ./cmd/storage
+    username:
+      from_secret: docker_username
+    password:
+      from_secret: docker_password
     dockerfile: cmd/storage/Dockerfile
-    group: docker
     repo: gitpods/storage
     tag:
     - latest
+  when:
+    branch:
+    - master
+    event:
+    - push
+
+depends_on:
+- testing
+
+---
+kind: pipeline
+name: ui
+
+platform:
+  os: linux
+  arch: amd64
+
+workspace:
+  base: /go
+  path: src/github.com/gitpods/gitpods
+
+steps:
+- name: download
+  image: google/dart:1.24
+  pull: always
+  commands:
+  - cd ui/
+  - pub get --no-precompile
+  volumes:
+  - name: pub
+    path: /root/.pub-cache
+
+- name: analyzer
+  image: google/dart:1.24
+  pull: always
+  commands:
+  - cd ui/
+  - dartanalyzer ./lib/
+  volumes:
+  - name: pub
+    path: /root/.pub-cache
+
+- name: dart
+  image: google/dart:1.24
+  pull: always
+  commands:
+  - cd ui/
+  - pub build
+  volumes:
+  - name: pub
+    path: /root/.pub-cache
+
+- name: cleanup
+  image: google/dart:1.24
+  pull: always
+  commands:
+  - cd ui/
+  - rm -rf build/web/packages
+  volumes:
+  - name: pub
+    path: /root/.pub-cache
+
+- name: bindata
```

Comment:
Same here, I don't see a real reason to split this up. IMHO we should simply do this packr command before running `go build`

Reply:
So far I have split that with the background to use a packr docker image to avoid downloading the release on every build.